www/docs/src/components/page-title/page-title.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from 'react'
+import { ChevronRight } from 'lucide-react'
+import { type Props } from '@astrojs/starlight/props'
+
+const PageTitle = ({ entry }: Props) => {
+  const [crumbs, setCrumbs] = useState<string[]>([])
+  const { data } = entry
+
+  useEffect(() => {
+    const formattedCrumbs = window.location.pathname
+      .replace(/^\/|\/$/g, '')
+      .split('/')
+    setCrumbs(formattedCrumbs)
+  }, [])
```

Comment:
We can achieve this without using `useEffect` or `useState` by passing in a prop from the astro component.

So in `astro-page-title.astro` we can pass in `crumbs={Astro.url.pathname.replace(/^\/|\/$/g, '').split('/')}` and then we also don't need `client:load` since the component can now be statically rendered.

```suggestion
import React from 'react'
import { ChevronRight } from 'lucide-react'
import { type Props } from '@astrojs/starlight/props'

const PageTitle = ({ entry, crumbs }: Props & { crumbs: string[] }) => {
  const { data } = entry
```